CHANGELOG.md

```diff
@@ -1,5 +1,441 @@
+The old names are still supported so as to not suddenly break existing code based on them, but they have now been deprecated.
+For example:
+
+```python
+from aiida.plugins import DataFactory
+Int = DataFactory('int')  # Old name
+Int = DataFactory('core.int')  # New name
+```
+
+Note that entry point names are also used on the command line. For example:
+
+```console
+$ verdi computer setup -L localhost -T local -S direct
+# now changed to
+$ verdi computer setup -L localhost -T local -S core.direct
+```
+
+### Improvements to the AiiDA storage architecture ♻️
+
+Full details on the AiiDA storage architecture are available in the [storage architecture section](docs/source/internals/storage/architecture.rst).
+
+The storage refactor incorporates four major changes:
+
+- The `django` and `sqlalchemy` storage backends have been merged into a single `psql_dos` backend (PostgreSQL + Disk-Objectstore).
+  - See the [`psql_dos` storage format](docs/source/internals/storage/psql_dos.rst) for details.
+  - This has allowed for the `django` dependency to be dropped.
+
+- The file system node repository has been replaced with an object store implementation.
+  - The object store automatically deduplicates files, and allows for the compression of many objects into a single file, thus significantly reducing the number of files on the file system and memory utilisation (by orders of magnitude).
+  - Note, to make full use of object compression, one should periodically run `verdi storage maintain`.
+  - See the [repository design section](docs/source/internals/storage/repository.rst) for details.
+
+- Command-line interaction with a profile's storage has been moved from `verdi database` to `verdi storage`.
+
+- The AiiDA archive format has been redesigned as the `sqlite_zip` storage backend.
+  - See the [`sqlite_zip` storage format](docs/source/internals/storage/sqlite_zip.rst) for details.
+  - The new format allows for streaming of data during exports and imports, significantly reducing both the time and memory utilisation of these actions.
+  - The archive can now be loaded directly as a (read-only) profile, without the need to import it first, see [this Jupyter Notebook tutorial](docs/source/howto/archive_profile.md).
+
+On first using `aiida-core` v2.0, your AiiDA configuration will be automatically migrated to the new version (this can be reverted by `verdi config downgrade`).
+To update existing profiles and archives to the new storage formats, simply use `verdi storage migrate`, or `verdi archive migrate`.
+
+:::{important}
+The migration of large storage repositories is a potentially time-consuming process.
+It may take several hours to complete, depending on the size of the repository.
+It is also advisable to make a full manual backup of any AiiDA setup with important data: see [the installation management section](docs/source/howto/installation.rst) for more information.
```

Comment:
Why not refer directly to the backup section?

Reply:
So the idea here, is that the Markdown links will work, when you are viewing the "raw" changelog file.

i.e. if you go to https://github.com/chrisjsewell/aiida_core/blob/v2-changelog/CHANGELOG.md

<img width="1034" alt="image" src="https://user-images.githubusercontent.com/2997570/158086011-2d6e3423-1fba-46e2-a7a1-2330355cb5bc.png">

You can click on the link, and it will actually take you to that documentation source page.

That would not work, if you simply use e.g. `` {ref}`the installation management section <how-to:installation:backup>` ``

In MyST you can actually link to heading anchors: https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#auto-generated-header-anchors,
BUT that only works if the "target" page is also Markdown,
e.g. `[the installation management section](docs/source/howto/installation.md#backing-up-your-installation)`